app/Photo.php

```diff
@@ -141,9 +148,9 @@ public function prepareData()
 		}
 
 		if ($this->medium2x != '') {
-			$url2x = explode('.', $this->url);
-			$url2x = $url2x[0].'@2x.'.$url2x[1];
-			$photo['medium2x'] = Config::get('defines.urls.LYCHEE_URL_UPLOADS_MEDIUM').$url2x;
+			$photoName2x = explode('.', $photoName);
+			$photoName2x = $photoName2x[0].'@2x.'.$photoName2x[1];
```

Comment:
this can be done around line 139. This avoids to do it twice: once for medium2x and once for small2x

Reply:
Right. I wrote it that way because all the 2x stuff is optional (and off by default) so I thought we shouldn't do it unless it's needed. But I agree that readability suffers as a result. I'll fix it.